app/modules/ws2812.c

```diff
@@ -14,30 +15,45 @@ typedef struct {
   uint8_t values[0];
 } ws2812_buffer;
 
-// Stream data using UART1 routed to GPIO2
-// NODE_DEBUG should not be activated because it also uses UART1
-static void ICACHE_RAM_ATTR ws2812_write(uint8_t pin, uint8_t *pixels, uint32_t length) {
-
-  // Data are sent LSB first, with a start bit at 0, an end bit at 1 and all inverted
-  // 0b00110111 => 110111 => [0]111011[1] => 10001000 => 00
-  // 0b00000111 => 000111 => [0]111000[1] => 10001110 => 01
-  // 0b00110100 => 110100 => [0]001011[1] => 11101000 => 10
-  // 0b00000100 => 000100 => [0]001000[1] => 11101110 => 11
-  uint8_t _uartData[4] = { 0b00110111, 0b00000111, 0b00110100, 0b00000100 };
-
+// Init UART1 to be able to stream WS2812 data
+// We use GPIO2 as output pin
+static void ICACHE_FLASH_ATTR ws2812_init() {
   // Configure UART1
   // Set baudrate of UART1 to 3200000
   WRITE_PERI_REG(UART_CLKDIV(1), UART_CLK_FREQ / 3200000);
   // Set UART Configuration No parity / 6 DataBits / 1 StopBits / Invert TX
   WRITE_PERI_REG(UART_CONF0(1), UART_TXD_INV | (1 << UART_STOP_BIT_NUM_S) | (1 << UART_BIT_NUM_S));
 
+  // Pull GPIO2 down
+  platform_gpio_mode(4, PLATFORM_GPIO_OUTPUT, PLATFORM_GPIO_FLOAT);
+  platform_gpio_write(4, 0);
+
+  // Waits 10us to simulate a reset
+  os_delay_us(10);
+
   // Redirect UART1 to GPIO2
   // Disable GPIO2
   GPIO_REG_WRITE(GPIO_ENABLE_W1TC_ADDRESS, BIT2);
   // Enable Function 2 for GPIO2 (U1TXD)
   PIN_FUNC_SELECT(PERIPHS_IO_MUX_GPIO2_U, FUNC_U1TXD_BK);
+}
+
+
+// Stream data using UART1 routed to GPIO2
+// ws2812.init() should be called first
+//
+// NODE_DEBUG should not be activated because it also uses UART1
+static void ICACHE_RAM_ATTR ws2812_write(uint8_t *pixels, uint32_t length) {
+
+  // Data are sent LSB first, with a start bit at 0, an end bit at 1 and all inverted
+  // 0b00110111 => 110111 => [0]111011[1] => 10001000 => 00
+  // 0b00000111 => 000111 => [0]111000[1] => 10001110 => 01
+  // 0b00110100 => 110100 => [0]001011[1] => 11101000 => 10
+  // 0b00000100 => 000100 => [0]001000[1] => 11101110 => 11
+  uint8_t _uartData[4] = { 0b00110111, 0b00000111, 0b00110100, 0b00000100 };
```

Comment:
`_uartData` is a local variable - is there a runtime penalty in case it's being initialized on the stack each time `ws2812_write()` is called?